cmd/mic/main.go

```diff
@@ -18,16 +18,27 @@ var (
 	forceNamespaced   bool
 	versionInfo       bool
 	syncRetryDuration time.Duration
+	leaderElectionCfg mic.LeaderElectionConfig
 )
 
 func main() {
 	defer glog.Flush()
+	hostName, err := os.Hostname()
+	if err != nil {
+		glog.Fatalf("Get hostname failure.")
+	}
 	flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to the kube config")
 	flag.StringVar(&cloudconfig, "cloudconfig", "", "Path to cloud config e.g. Azure.json file")
 	flag.BoolVar(&forceNamespaced, "forceNamespaced", false, "Forces namespaced identities, binding, and assignment")
 	flag.BoolVar(&versionInfo, "version", false, "Prints the version information")
 	flag.DurationVar(&syncRetryDuration, "syncRetryDuration", 3600*time.Second, "The interval in seconds at which sync loop should periodically check for errors and reconcile.")
 
+	// Leader election parameters
+	flag.StringVar(&leaderElectionCfg.ResourceName, "leader-election-name", hostName, "leader name. default is 'hostname'")
```

Comment:
default shouldn't be host name, default should be a well known string. Mic pods will get different host names.

Reply:
isn't that for ID ? Shouldn't the resource name be unique ?

Reply:
`leaderElectionCfg.ResourceName` throw me off. i thought it is for endpoint name, not for the identity of the leader. Let us try to avoid the same for future users. 
- rename the field ResourceName to leader identity or something of that sort. 
- rename field `ID` to `Identity`

Reply:
ResourceName ==> Instance
ID ==> Name